2024/XenSummit/Challenges_and_Status_of_Enabling_TrenchBoot_in_Xen_Hypervisor.md

```diff
@@ -0,0 +1,234 @@
+  confirm whether the device has been tampered with or not
+- Initially only Intel TXT and TPM1.2 with tboot was supported using legacy
+  BIOS boot mode, now TrenchBoot replaces tboot, extending the support to
+  TPM2.0 and AMD SKINIT
+
+---
+
+# Demo: Qubes OS Anti Evil Maid
+
+.center[ <a href="https://asciinema.org/a/661348?cols=96&rows=24"
+target="_blank"> <img src="https://asciinema.org/a/661348.svg" width="400px"/>\
+</a> ]
+
+.footnote[
+Watch on [asciinema](https://asciinema.org/a/661348?cols=96&rows=24)
+]
+
+---
+
+# Current status of the work
+
+- Added TPM 2.0 support for dom0 kernel and initrd measurements
+- Added eventlog extraction and parsing
+- Rebased changes onto the most recent TrenchBoot support patches to Linux
+  kernel introducing a unified TrenchBoot boot protocol
+- Final shape of patches is not yet known, the work on finalizing the boot
+  protocol including UEFI is pending on the [Linux kernel mailing list,
+  currently patchset
+  v8](https://lore.kernel.org/lkml/8d543a15-af62-4403-b2e0-3b395edfe9e4@amd.com/T/)
+- Simplified installation of the required packages by providing a RPM
+  repository for QubesOS
+- Implemented support for booting with Intel TXT and TPM 2.0 in legacy BIOS
+  boot mode
+- Implemented support for booting with AMD SKINIT and TPM 1.2/2.0 in legacy
+  BIOS boot mode
+
+---
+
+# Current ongoing efforts
+
+- Provide package repositories for other major distros (Debian, Ubuntu Fedora)
+- Prepare tools for assessing the readiness of the system and BIOS to perform
+  Dynamic Launch
+- Adding DRTM Ready indicator to the system security reports: QubesOS Security
+  Report, [fwupd HSI](https://fwupd.github.io/libfwupdplugin/hsi.html)
+
+.center[.image-40[![](/img/qubes-sec-report.png)]]
+
+---
+
+# Try TenchBoot
+
+It is relatively easy to get a hardware which supports DRTM:
+
+- Intel-based tested and known to work:
+  * Protectli VP4670
+  * HP EliteDesk 800 G2 Desktop Mini
+- Any Intel vPro laptop/desktop should generally work, but DRTM is sensitive
+  to BIOS bugs related to Intel TXT initialization (some platforms simply do
+  not set it up properly)
+
+- AMD-based
+  * Terminal HP T630
+- Almost any pre-Zen3 should work out of the box
+
+---
+
+# Xen as DLME for UEFI
+
+### Future planned work
+
+- Xen will be able to initiate the DL Event via the code module exposed by DCE
+  Preamble (GRUB)
+  * Although Xen does not use that much information from UEFI we want to unify
+    the flow with TrenchBoot implementation for Linux kernel
+  * Native EFI PE entry point shall be supported for UEFI Secure Boot purposes
+- Xen will need to obtain the module from Secure Launch Resource Table linked
+  in the EFI System Table Configuration Tables
+- Xen will fill the necessary information for the DCE (like Xen address in
+  memory to be measured) and call the module
+
+More details on [trenchboot.org](https://trenchboot.org/specifications/Secure_Launch/)
```

Comment:
This should be presented in form of roadmap which could be revisited in future.

Reply:
974a00d880ea18219c3981708c8dc457a1209b4f